### PR TITLE
fix(android/engine): Revise updateOldKeyboardsList

### LINKED ISSUE #3134: [Android] Bring back KMManager.updateOldKeyboardsList()
During the refactoring in #3113, KMManager.updateOldKeyboardsList() was removed.

While there **shouldn't** be any Keymean installs still using `us` or `european2`keyboards, we should bring back updateOldKeyboardsList() to migrate the keyboards.

Depends on Cloud keyboard updates to install keyboard package before the keyboard info can be migrated from "cloud" to package ID.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #3134

This PR revises updateOldKeyboardsList() to migrate the legacy keyboards list (HashMap format) into the new JSON list of keyboards.

Based on the previous updateOldKeyboardsList(), it now does the following:

Loop through keyboards list:
1. If the keyboard ID is `sil_euro_latin` or any of its deprecated keyboards (`us`, `european`, `european2`), add the default `sil_euro_latin` keyboard (once)
2. If package undefined, set to "cloud" and update entry
3. If version undefined, get the latest from the file version and update entry

Also included a unit test.

Only handling deprecated keyboards for sil_euro_latin since we know the info.
Other migration will be on a separate PR.

### CODE REVIEW

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/KeyboardController.java [hunk 1]
```diff
@@ -63,27 +63,7 @@ public synchronized void initialize(Context context) {
...
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_CustomHelpLink, ""),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_KMPLink, ""),
-              isNewKeyboard,
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_Font, null),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_OskFont, null)
-            );
-            list.add(k);
-          }
-
+          KMManager.updateOldKeyboardsList(context, dat_list, list);
```
Comment (jahorton) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/KeyboardController.java:66]: Again, why are we passing `list` in as a parameter?  Examining the code above, we know that its value is guaranteed to be an empty list instance; `updateOldKeyboardsList` neither needs nor cares for the parameter's value except as an output.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java [hunk 2]
```diff
@@ -719,6 +720,66 @@ public boolean accept(File pathname) {
...
+   */
+  public static void updateOldKeyboardsList(Context context, ArrayList<HashMap<String, String>> dat_list,
+                                            List<Keyboard> list) {
+    if (dat_list == null || list == null) {
+      return;
+    }
+
+    boolean defaultKeyboardInstalled = false;
+    for(HashMap<String, String> kbdMap : dat_list) {
+      boolean isNewKeyboard = kbdMap.containsKey(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD) &&
```
Comment (jahorton) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:744]: Sorry if this question has been answered before elsewhere, but... what is this line about?  The "internal new keyboard" thing?  Why would it vary between "true" and "false" as part of the resource migration process?

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:744]: The "internal new keyboard" flag is used after installing a keyboard to highlight the keyboard in the keyboard list with a "[new]" string. When the keyboard gets switched, the flag is removed.

I can simplify things by just setting this to false during the keyboard list migration.

**Comment by darcywong00** [comment]
>  How does this "revise" updateOldKeyboardsList?
> Also, why is the new method on KMManager instead of within KeyboardController?

Some of the history is in the issue description above.
`KMManager.updateOldKeyboardsList()` used to exist until 14.0 development a few months ago. This PR "revises" updateOldKeyboardsList by bringing the method back to KMManager. Just not exactly the same as it used to be.

**Pull request merged**